src/main/java/com/hashedin/bitly/ShortCodeMappingService.java

```diff
@@ -0,0 +1,63 @@
+package com.hashedin.bitly;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ShortCodeMappingService {
+
+	private final Map<String, URI> mapping;
+	
+	public ShortCodeMappingService(@Value("classpath:mapping.properties") Resource mappingResource) throws IOException {
+		this.mapping = loadMapping(mappingResource);
+	}
+	
+	private static final Map<String, URI> loadMapping(Resource mappingResource) throws IOException {
+		Properties props = new Properties();
+		try(InputStream in = mappingResource.getInputStream()) {
+			props.load(mappingResource.getInputStream());			
+		}
+		
+		Map<String, URI> mapping = new HashMap<String, URI>();
+		
+		for (Entry<Object, Object> entry : props.entrySet()) {
+			try {
+				String key = entry.getKey().toString();
+				URI uri = new URI(entry.getValue().toString());
+				validateUri(uri);
+				mapping.put(key, uri);
```

Comment:
A Properties object is convenient to read, but difficult to work with because keys and values are objects. So we convert it into a Map<String, URI>, and along the way validate the URIs.

In general, you should perform your validations as early as possible. This code is executing at startup, and so if a URI is invalid, it will prevent the application from loading. This is called fail fast, and it is a good thing.

Even more general, compile time errors are better than error during loading, which in turn is better than errors at execution time.